### ISSUE TITLE
dialects (memref): Parsing scalar memrefs fails with declarative syntax

### ISSUE THREAD

**Issue by compor** [xdslproject/xdsl#2063]
MWE:

```mlir
%arg = "test.op"() : () -> memref<f64>
%0 = "test.op"() : () -> f64
%1 = memref.load %arg[] : memref<f64>
memref.store %0, %arg[] : memref<f64>
```

`xdsl-opt` complains that operand is expected for `%arg[]`.

`git bisect` says this was introduced in 215c88f63657b560759a3dd62368dc5ffefc29d8 (or PR #2023), so it seems like a regression.
Surprisingly, we have no similar filechecks for this ATM.

**Comment by compor** [comment]
CC @PapyChacal

**Comment by compor** [comment]
Resolved in #2069
